src/curve/constant-product/ConstantProductPair.sol

```diff
@@ -77,18 +77,15 @@ contract ConstantProductPair is ReservoirPair {
 
         if (lFeeOn) {
             uint256 lSqrtOldK = FixedPointMathLib.sqrt(kLast); // gas savings
+            uint256 lSqrtNewK = FixedPointMathLib.sqrt(aReserve0 * aReserve1);
 
-            if (lSqrtOldK != 0) {
-                uint256 lSqrtNewK = FixedPointMathLib.sqrt(aReserve0 * aReserve1);
+            if (lSqrtNewK > lSqrtOldK) {
```

Comment:
What happens in the scenario where we turn platform fee off? Won't platform fee be charged on the full range because sqrt_old_k is zero?

Reply:
`kLast` is always written to at the end of mint+burn regardless if platform fee is on or off. 

There is a check of `feeOn` at the beginning of the function. So if platform fee was off, this wouldn't even happen. So the scenario you described will not take place

Reply:
If we consider more extreme scenarios, such as turning platform fee off and then on again, it would also work properly since the last invariant is always written to